### PR TITLE
Electron stream plugin fix

### PR DESCRIPTION
**Pull request by shervin86**
This fixes the absence of the SelectedElectronFEDListProducer after the merge of PR #8306 
The .cc and .h of the plugin has been moved to the plugin directory and now compiles.
Now the plugin is correctly restored as in any 75X pre release before #8306.

To avoid conflicts I have already merged with #9112.
This can PR can be merged after that or at the place of.

@davidlange6 @rgerosa @amassiro

### CODE REVIEW

**Comment by shervin86** [comment]
I have fixed it. I don't know why when I had compiled with scram I got no errors...
Now it compiles with `scram b vclean && scram b -k -j 8`

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
+1

**Comment by shervin86** [comment]
@davidlange6  @nclopezo @smuzaffar can this be merged to see that on the next IB and RelVals. Is there any concern about this?

File: Calibration/EcalAlCaRecoProducers/plugins/BuildFile.xml [hunk 1]
```diff
@@ -8,4 +8,15 @@
...
+<use   name="CalibFormats/SiStripObjects"/>
+<use   name="CalibCalorimetry/HcalAlgos"/>
+<use   name="CondFormats/EcalObjects"/>
+<use   name="CondFormats/DataRecord"/>
+<use   name="CondFormats/HcalObjects"/>
+<use   name="CondFormats/SiPixelObjects"/>
+<use   name="Geometry/CaloGeometry"/>
+<use   name="RecoEcal/EgammaCoreTools"/>
+<use   name="TrackingTools/TrackAssociator"/>
 <flags   EDM_PLUGIN="1"/>
```
Comment (davidlange6) [inline Calibration/EcalAlCaRecoProducers/plugins/BuildFile.xml]: was trying to figure out why all of these dependencies came in. Could you clean up the header file for the moved plugin to use only the incldues that it needs (which is almost none of them- use "class blah;" instead) and add the minimal set of headers to the cc file.

Comment (shervin86) [inline Calibration/EcalAlCaRecoProducers/plugins/BuildFile.xml]: Dear David,
I'm cleaning it a bit, but a large number of includes are needed and the BuildFile will not be reduced.
Are you suggesting to replace 
use name= with  use class= ?
Can you point me to an example please?

Thanks
-- 
 							Shervin

Comment (davidlange6) [inline Calibration/EcalAlCaRecoProducers/plugins/BuildFile.xml]: http://stackoverflow.com/questions/8632430/about-class-declaration-in-header-file-c

**Comment by shervin86** [comment]
Since without this fix, the ElectronStream cannot be integrated in the HLT menu, can we merge this PR and have another one to fix the dependencies? I have already prepared a new branch, but I think we still need some iteration to have it in good shape

**Comment by davidlange6** [comment]
@shervin86 - would rather get this fixed (as things tend not to get fixed after) - what is the problem you are encountering?

**Comment by shervin86** [comment]
It would be very kind of you if you can explain better the fixes you are asking in person.
Can we meet today?

-- 
 							Shervin

**Comment by shervin86** [comment]
Branch updated removing includes from the header file of the plugin and moved to the cc.
The BuildFile has been cleaned reducing the number of dependencies.
The branch has been rebased w.r.t. to the HEAD of CMSSW_7_5_X
Thanks to @davidlange6 for the help

**Comment by shervin86** [comment]
@amassiro @rgerosa @emanueledimarco have requested to follow the PR

**Comment by shervin86** [comment]
Can you please trigger the tests (if still needed)?

**Comment by mmusich** [comment]
please test

**Comment by shervin86** [comment]
Test passed

**Comment by mmusich** [comment]
+1

**Comment by shervin86** [comment]
@davidlange6 @degano @smuzaffar can this be merged?

**Comment by emanueledimarco** [comment]
In order to have the #9085 integrated, can we merge this, when possible?

**Comment by davidlange6** [comment]
+1
Done - but for the backport please also remove the try -catch blocks if possible (I think it is)

**Pull request merged**